sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -150,8 +162,27 @@ class RocksDB(
           metadata.numKeys
         }
         numKeysOnWritingVersion = numKeys
-        numKeysOnLoadedVersion = numKeys
-
+        // Replay change log from the last snapshot to the loaded version.
+        // This will be noop if changelog checkpointing is disabled.
+        for (v <- latestSnapshotVersion + 1 to version) {
```

Comment:
Prefer to embed this in an if condition for easier readability

Reply:
Refactored.